happi/cli.py

```diff
@@ -97,223 +47,314 @@ def happi_cli(args):
                         fmt='[%(asctime)s] - %(levelname)s -  %(message)s')
     logger.debug("Set logging level of %r to %r", shown_logger.name, level)
 
-    # Version endpoint
-    if args.version:
-        print(f'Happi: Version {happi.__version__} from {happi.__file__}')
-        return
-    logger.debug('Command line arguments: %r' % args)
-
-    client = happi.client.Client.from_config(cfg=args.path)
+    # gather client
+    client = happi.client.Client.from_config(cfg=path)
     logger.debug("Happi client: %r" % client)
-    logger.debug('Happi command: %r' % args.cmd)
 
-    if args.cmd == 'search':
-        logger.debug("We're in the search block")
+    # insert items into context to be passed to subcommands
+    ctx.obj = client
```

Comment:
The `ctx.obj` attribute is special in `click`.  It exists on the context object as packaged, and is meant to hold whatever extra object/state you want to pass around.  I'm not sure why you can't specify your own attributes (suspect it's because context objects are created and destroyed, and obj is passed along)

Reply:
Interesting - I think for our future selves it might be worthwhile making what you just said a comment about `obj`?